File: Data Structures & Algorithms/duplicate-integer/submission-0.py
class Solution:
    def hasDuplicate(self, nums: List[int]) -> bool:
        # if num in num_dict - > return true else add num to num_dict
        num_dict = {}
        for num in nums:
            if num in num_dict:
                return True
            num_dict[num] = 0
        return False

File: Data Structures & Algorithms/kth-largest-integer-in-a-stream/submission-0.cpp
class KthLargest {
public:
    priority_queue<int, vector<int>, greater<int>> minHeap;
    int limit;
    KthLargest(int k, vector<int>& nums) {
        limit = k;
        for (int num : nums) {
            minHeap.push(num);
            if (minHeap.size() > k) minHeap.pop();
        }
    }
    
    int add(int val) {
        minHeap.push(val);
        if (minHeap.size() > limit) minHeap.pop();
        return minHeap.top();
        
    }
};
